app/domain/authentication/authn_jwt/identity_from_url_provider.rb

```diff
@@ -0,0 +1,18 @@
+module Authentication
+  module AuthnJwt
+    # Provides jwt identity from information in the URL
+    class IdentityFromUrlProvider < IdentityProviderInterface
+      def initialize(authentication_parameters)
+        @authentication_parameters = authentication_parameters
+      end
+
+      def provide_jwt_identity
+        @authentication_parameters.username
+      end
+
+      def identity_available?
+        !@authentication_parameters.username.nil?
```

Comment:
can it be empty string ?

Reply:
Its already tested. If its nil function return false

Reply:
I am asking why the code is `.nil?` and not `.blank?` (which is nil and empty) ?

Reply:
Changed to blank. good catch I just now read blank also returns true if the string is just whitespaces. A case we want to abort and I will add it to the tests